src/components/tables/TableSort.tsx

```diff
@@ -0,0 +1,58 @@
+// tslint:disable-next-line:no-unused-variable
+import * as React from 'react';
+import { ITableHeaderCellProps } from './TableHeaderCell';
+import { Svg } from '../svg/Svg';
+
+export enum SortTypes {
+  Ascending,
+  Descending,
+  None
+}
+
+export interface ITableSortProps extends ITableHeaderCellProps {
+  sortType: SortTypes;
+  isFixedHeader: boolean;
+}
+
+export abstract class TableSort implements ITableHeaderCellProps {
```

Comment:
Mostly because I did not want to break the current framework: the class `TableHeader` takes an array of `ITableHeaderCellProps` and instantiates a cell with every item in this array.

I had 2 options: modify the `ITableHeaderProps` so that `columns` is now an array of TableHeaderCell or creating a class that computes the properties of the cells. 
The sort component content different when the table is fixed or not, the class of the cell is different if the sort type changes, meaning I could define my cells as follow:

    title: 'Name',
    className: 'admin-sort admin-sort-ascending', // need to know all class names or code something to convert it
    cellContent: <TableSort/>

or like this:

    new NameSort(SortTypes.Ascending);

The implementation on my side has some classes `SelectedSort`, `FieldNameSort`, with just a constructor which takes a `SortType` + onClick function as input and fixes the `title` by itself, `className` is computed by calling the base class.

Looks quite clean to me. Anything better to suggest?